### PR TITLE
Single video page

### PR DESCRIPTION
**Pull request by AtulPant2704**
- User can like or add video to playlist from Single video page
`Note:- Watch later won't work as of now, will work on it, in the end.`

Preview:- https://deploy-preview-7--checkmate-stream.netlify.app/explore

### CODE REVIEW

File: src/pages/SingleVideoPage/SingleVideoPage.jsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+import { useLikes, useAuth, usePlaylistModal } from "../../hooks";
+import { addToLikesHandler, removeFromLikesHandler } from "../../utils";
+import { getSingleVideoHandler } from "../../utils";
+
+const SingleVideoPage = () => {
+    const navigate = useNavigate();
+    const [video, setVideo] = useState({});
+    const { videoID } = useParams();
+    const { authState: { token } } = useAuth();
+    const { likesState: { likes }, likesDispatch } = useLikes();
```
Comment (kart-c) [inline src/pages/SingleVideoPage/SingleVideoPage.jsx:13]: like the destructuring here 👍

**Review by AnkurChunekar — APPROVED** [review]
Single page UI looks good, code is neat and clean great work @AtulPant2704

**Comment by kart-c** [comment]
Video is getting liked from the single video page but the app throws an error when I unlike from the likes page. Can you fix this? Also, your explore section card buttons need to `stopPropagation`. Liking/adding to the playlist  the card from the explore section opens the single video page

**Comment by starver20** [comment]
Screenshot_20220329_222538
Deleting video from liked videos is throwing error, other functionalities are working as expected

**Comment by AtulPant2704** [comment]
> Video is getting liked from the single video page but the app throws an error when I unlike from the likes page. Can you fix this? Also, your explore section card buttons need to `stopPropagation`. Liking/adding to the playlist the card from the explore section opens the single video page

Thankyou for pointing out that bug, I have changed it now.

**Comment by AtulPant2704** [comment]
> Screenshot_20220329_222538 Deleting video from liked videos is throwing error, other functionalities are working as expected

Fixed that now.

**Comment by kart-c** [comment]
It's working fine now. Good work.

**Review by kart-c — APPROVED** [review]

**Comment by starver20** [comment]
> > Screenshot_20220329_222538 Deleting video from liked videos is throwing error, other functionalities are working as expected
> 
> Fixed that now.

Delete function is working fine now, but now after deleting it is redirecting to single video page, you should add e.stopPropagation on that click to avoid that.

**Comment by AtulPant2704** [comment]
> > > Screenshot_20220329_222538 Deleting video from liked videos is throwing error, other functionalities are working as expected
> > 
> > 
> > Fixed that now.
> 
> Delete function is working fine now, but now after deleting it is redirecting to single video page, you should add e.stopPropagation on that click to avoid that.

Fixed that.

**Review by starver20 — APPROVED** [review]

**Pull request merged**